package.json

```diff
@@ -516,7 +528,38 @@
                     "name": "Targets"
                 }
             ]
-        }
+        },
+        "viewsWelcome": [

```

Comment:
Hmm, I probably don't know enough about how extension views are registered. But, this seems like quite a bit of UI definition for this view inside the package.json. I don't see the equivalent for the list of targets, and I wonder if that's in it's own file? or handled in extensions.ts with the CDPTargetsProvider? Anyway. I can see this quickly growing in length if we add more to the sidebar view, and I wonder if there is a way to move the UI definitions to it's own file?

Reply:
Feel free to ignore this if it doesn't make sense to do that

Reply:
I think we will need to keep it in package.json because viewsWelcome is a contribution point specifically defined for VSCode extensions (more about that [here](https://code.visualstudio.com/api/references/contribution-points#contributes.viewsWelcome))

Quick rundown though: the sidepane view is registered in package.json as `vscode-edge-devtools-view.targets`, then in extension.js the sidepane view subscribes to CDPTargetsProvider, which has the logic to get whatever info VSCode needs to display the targets. viewWelcome only shows when the list that it's view subscribes to is empty, plus some additional conditions (in the `"when"` property). So there's a lot more UI definition for viewsWelcome vs the targets view in package.json because viewsWelcome is more static and the target list is more dynamic.

Reply:
Ahh, totally makes sense. Thanks for the explanation and docs!